File: native/app/src/main/java/com/example/anative/models/Expense.java
package com.example.anative.models;

public class Expense {
    private long id;
    private long projectId;
    private String date;
    private double amount;
    private String currency;
    private String type;
    private String paymentMethod;
    private String claimant;
    private String paymentStatus;
    private String description;
    private String location;

    public Expense() {}

    public Expense(long id, long projectId, String date, double amount, String currency,
                   String type, String paymentMethod, String claimant, String paymentStatus,
                   String description, String location) {
        this.id = id;
        this.projectId = projectId;
        this.date = date;
        this.amount = amount;
        this.currency = currency;
        this.type = type;
        this.paymentMethod = paymentMethod;
        this.claimant = claimant;
        this.paymentStatus = paymentStatus;
        this.description = description;
        this.location = location;
    }

    public Expense(long projectId, String date, double amount, String currency,
                   String type, String paymentMethod, String claimant, String paymentStatus,
                   String description, String location) {
        this.projectId = projectId;
        this.date = date;
        this.amount = amount;
        this.currency = currency;
        this.type = type;
        this.paymentMethod = paymentMethod;
        this.claimant = claimant;
        this.paymentStatus = paymentStatus;
        this.description = description;
        this.location = location;
    }

    public long getId() { return id; }
    public void setId(long id) { this.id = id; }

    public long getProjectId() { return projectId; }
    public void setProjectId(long projectId) { this.projectId = projectId; }

    public String getDate() { return date; }
    public void setDate(String date) { this.date = date; }

    public double getAmount() { return amount; }
    public void setAmount(double amount) { this.amount = amount; }

    public String getCurrency() { return currency; }
    public void setCurrency(String currency) { this.currency = currency; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public String getPaymentMethod() { return paymentMethod; }
    public void setPaymentMethod(String paymentMethod) { this.paymentMethod = paymentMethod; }

    public String getClaimant() { return claimant; }
    public void setClaimant(String claimant) { this.claimant = claimant; }

    public String getPaymentStatus() { return paymentStatus; }
    public void setPaymentStatus(String paymentStatus) { this.paymentStatus = paymentStatus; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
}

File: cross/lib/models/expense.dart
class Expense {
  final String id;
  String? projectId;
  String expenseCode;
  String date;
  double amount;
  String currency;
  String type;
  String paymentMethod;
  String claimant;
  String paymentStatus;
  String description;
  String location;

  Expense({
    required this.id,
    this.projectId,
    required this.expenseCode,
    required this.date,
    required this.amount,
    required this.currency,
    required this.type,
    required this.paymentMethod,
    required this.claimant,
    required this.paymentStatus,
    required this.description,
    required this.location,
  });

  factory Expense.fromMap(String id, Map<String, dynamic> map) {
    return Expense(
      id: id,
      projectId: map['projectId']?.toString(),
      expenseCode: map['expenseCode'] ?? '',
      date: map['date'] ?? '',
      amount: (map['amount'] ?? 0.0).toDouble(),
      currency: map['currency'] ?? '',
      type: map['type'] ?? '',
      paymentMethod: map['paymentMethod'] ?? '',
      claimant: map['claimant'] ?? '',
      paymentStatus: map['paymentStatus'] ?? '',
      description: map['description'] ?? '',
      location: map['location'] ?? '',
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'projectId': projectId,
      'expenseCode': expenseCode,
      'date': date,
      'amount': amount,
      'currency': currency,
      'type': type,
      'paymentMethod': paymentMethod,
      'claimant': claimant,
      'paymentStatus': paymentStatus,
      'description': description,
      'location': location,
    };
  }
}